docs/FeatureManagement/FeatureFlag.v2.0.0.schema.json

```diff
@@ -5,7 +5,8 @@
   "type": "object",
   "title": "An Azure App Configuration Feature Declaration",
   "required": [
-    "id"
+    "id",
+    "enabled"
```

Comment:
I think all FM libraries treat this as optional today. Are we going to take a breaking change?

Reply:
Our portal has already treated `enabled` as required. (Portal/AppConfigPortalExtension/AppConfigPortalExtension/Client/Shared/JsonSchemas.ts Line200)

I think this is a bug. A feature flag without `enabled` looks like an invalid use case. 

> Are we going to take a breaking change?

IMO, yes. Microsoft schema support is still under preview. It is a good chance to take breaking change.

Reply:
Portal is just one of tools to create FF. FM libraries are the ones to enforce it.

I don't disagree with you. I would make the "enabled" property required if we started from scratch. Without FM libraries enforcing it, it doesn't matter what schema says. Here are two options.

**Option 1**
- Make "enabled" required in schema.
- .NET/Spring FM libraries continue to treat it as optional for compatibility.
- Python/JS FM libraries throw if the "enabled" property is absent.

**Option 2**
- Leave "enabled" as optional in schema.
- .NET/Spring FM libraries continue to treat it as optional.
- Python/JS FM libraries treat a feature flag as disabled if the "enabled" property is absent (the same behavior as .NET/Spring).

Reply:
> I would make the "enabled" property required if we started from scratch. 

In this case, I think option 2 is fine.

@jimmyca15 @rossgrambo @Eskibear @mrm9084 What do you think?

Reply:
Enabled already defaults to false in this schema- so calling it required here has no impact unless we removed that as well. I prefer removing it here and leaving the default.

Reply:
I agree with @rossgrambo

Reply:
enabled has a default value. It doesn't make sense to have a default value and be required.

I don't see any reason to make it required.

Reply:
> enabled has a default value.

I missed that. In this case, `enabled` should be required.